### PR TITLE
[18Dixie] ICG/SCL mergers

### PR DESCRIPTION
**Pull request by ventusignis**
I left this unfinished last November and picked this back up this week, my Ruby is a little rusty
### Implementation Notes

* **Explanation of Change**
Implement 18Dixie's ICG/SCL merger 

* **Screenshots**
image
image
image
image

* **Any Assumptions / Hacks**
No hacks for this

### CODE REVIEW

File: lib/engine/train.rb [hunk 1]
```diff
@@ -30,9 +30,10 @@ def initialize(name:, distance:, price:, index: 0, **opts)
       @obsolete = false
       @operated = false
       @ever_operated = false
-      @events = (opts[:events] || []).select { |e| @index == (e['when'] || 1) - 1 }
+      @events = select_events(opts[:events])
       @reserved = opts[:reserved] || false
       @requires_token = opts[:requires_token].nil? ? true : opts[:requires_token]
+      @variant_event_handling = opts[:variant_event_handling] || :default
```
Comment (ventusignis) [inline lib/engine/train.rb:36]: I'm going to revert the changes to lib/engine/train.rb

Comment (ventusignis) [reply lib/engine/train.rb]: Reverted

File: lib/engine/game/g_18_dixie/step/bankrupt.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+require_relative '../../../step/bankrupt'
+
+module Engine
+  module Game
+    module G18Dixie
+      module Step
+        class Bankrupt < Engine::Step::Bankrupt
+          def active?
+            active_entities.any?
```
Comment (tobymao) [inline lib/engine/game/g_18_dixie/step/bankrupt.rb:11]: !active_entities.empty?

File: lib/engine/game/g_18_dixie/step/icgscl_merge_consent.rb [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+            return false unless entity.corporation?
+            return false unless @round.merge_consent_merging_corp
+            return false if @round.merge_consent_pending_corps.empty?
+            return false unless entity == @round.merge_consent_pending_corps[0]
+
+            true
+          end
+
+          def help
+            return "Choosing to merge WILL form the #{merger_corp_name}" if @round.merge_consent_pending_corps.length < 2
```
Comment (tobymao) [inline lib/engine/game/g_18_dixie/step/icgscl_merge_consent.rb:79]: change all places where you use length with size

File: lib/engine/game/g_18_dixie/step/icgscl_merge_consent.rb [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+          def actions(entity)
+            return %w[choose] if turn_to_choose(entity)
+
+            []
+          end
+
+          def show_other
+            [@round.merge_consent_primary_corp,
+             @round.merge_consent_secondary_corp,
+             @round.merge_consent_merging_corp].freeze - [current_entity]
```
Comment (tobymao) [inline lib/engine/game/g_18_dixie/step/icgscl_merge_consent.rb:66]: what's the point of the freeze here?

File: lib/engine/game/g_18_dixie/step/bankrupt.rb (hunk 2 above)
Comment (ventusignis) [reply lib/engine/game/g_18_dixie/step/bankrupt.rb:11]: done

File: lib/engine/game/g_18_dixie/step/icgscl_merge_consent.rb (hunk 3 above)
Comment (ventusignis) [reply lib/engine/game/g_18_dixie/step/icgscl_merge_consent.rb:79]: done

File: lib/engine/game/g_18_dixie/step/icgscl_merge_consent.rb (hunk 4 above)
Comment (ventusignis) [reply lib/engine/game/g_18_dixie/step/icgscl_merge_consent.rb:66]: When I looked up the behavior of "array - array" the site I used had me think that the first array may be mutated. I've removed the `.freeze`
